src/tools/crawling_tools.py

```
@@ -0,0 +1,702 @@
+        scores = model.predict(pairs)
+
+        # Add scores to results and sort by score (descending)
+        for i, result in enumerate(results):
+            result["rerank_score"] = float(scores[i])
+
+        # Sort by rerank score
+        reranked = sorted(results, key=lambda x: x.get("rerank_score", 0), reverse=True)
+
+        return reranked
+    except Exception as e:
+        logger.error(f"Error during reranking: {e}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)